game-core/src/main/java/games/strategy/triplea/ui/BattlePanel.java

```diff
@@ -240,57 +240,50 @@ public void showBattle(
-                    isAmphibious,
-                    battleType,
-                    amphibiousLandAttackers);
-            battleFrame.setTitle(
-                attacker.getName()
-                    + " attacks "
-                    + defender.getName()
-                    + " in "
-                    + location.getName());
-            battleFrame.getContentPane().removeAll();
-            battleFrame.getContentPane().add(battleDisplay);
-            battleFrame.setMinimumSize(new Dimension(800, 600));
-            battleFrame.setLocationRelativeTo(JOptionPane.getFrameForComponent(BattlePanel.this));
-            PbemDiceRoller.setFocusWindow(battleFrame);
-            boolean foundHumanInBattle = false;
-            for (final Player gamePlayer :
-                getMap().getUiContext().getLocalPlayers().getLocalPlayers()) {
-              if ((gamePlayer.getGamePlayer().equals(attacker)
-                      && gamePlayer instanceof TripleAPlayer)
-                  || (gamePlayer.getGamePlayer().equals(defender)
-                      && gamePlayer instanceof TripleAPlayer)) {
-                foundHumanInBattle = true;
-                break;
-              }
-            }
-            if (ClientSetting.showBattlesWhenObserving.getValueOrThrow() || foundHumanInBattle) {
-              battleFrame.setAlwaysOnTop(true);
-              battleFrame.setVisible(true);
-              battleFrame.validate();
-              battleFrame.invalidate();
-              battleFrame.repaint();
-            } else {
-              battleFrame.setVisible(false);
+          battleDisplay =
+              new BattleDisplay(
+                  getData(),
+                  location,
+                  attacker,
+                  defender,
+                  attackingUnits,
+                  defendingUnits,
+                  killedUnits,
+                  attackingWaitingToDie,
+                  defendingWaitingToDie,
+                  BattlePanel.this.getMap(),
+                  isAmphibious,
+                  battleType,
+                  amphibiousLandAttackers);
+          battleFrame.setTitle(
+              attacker.getName() + " attacks " + defender.getName() + " in " + location.getName());
+          battleFrame.getContentPane().removeAll();
+          battleFrame.getContentPane().add(battleDisplay);
+          battleFrame.setMinimumSize(new Dimension(800, 600));
+          battleFrame.setLocationRelativeTo(JOptionPane.getFrameForComponent(BattlePanel.this));
+          PbemDiceRoller.setFocusWindow(battleFrame);
+          boolean foundHumanInBattle = false;
+          for (final Player gamePlayer :
+              getMap().getUiContext().getLocalPlayers().getLocalPlayers()) {
+            if ((gamePlayer.getGamePlayer().equals(attacker) && gamePlayer instanceof TripleAPlayer)
+                || (gamePlayer.getGamePlayer().equals(defender)
+                    && gamePlayer instanceof TripleAPlayer)) {
```

Comment:
Just a random thought:
If the instanceof check came before the equality check that would be a sneaky built-in null check as well